processing/src/main/java/io/druid/output/ByteBufferOutputBytes.java

```diff
@@ -0,0 +1,228 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.output;
+
+import com.google.common.base.Preconditions;
+import com.google.common.io.ByteSource;
+import com.google.common.primitives.Ints;
+import io.druid.io.ByteBufferInputStream;
+import io.druid.io.Channels;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.ByteBuffer;
+import java.nio.channels.WritableByteChannel;
+import java.util.ArrayList;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public abstract class ByteBufferOutputBytes extends OutputBytes
```

Comment:
Would you add unit tests for the HeapByteBufferOutputBytes and DirectByteBufferOutputBytes?

Reply:
Added tests